scripts/staging-ci/check-squash-into-staging.js

```diff
@@ -0,0 +1,43 @@
+'use strict'
+
+const { initGitConfig, executeCommand, printLog, printError, logAndExit } = require('../utils')
+
+const REPOSITORY = process.env.GIT_REPOSITORY
+const CI_COMMIT_SHA = process.env.CI_COMMIT_SHA
+const CI_COMMIT_SHORT_SHA = process.env.CI_COMMIT_SHORT_SHA
+const CI_COMMIT_REF_NAME = process.env.CI_COMMIT_REF_NAME
+const MAIN_BRANCH = process.env.MAIN_BRANCH
+
+async function main() {
+  await initGitConfig(REPOSITORY)
+  await executeCommand(`git fetch --no-tags origin ${MAIN_BRANCH}`)
+  const ciConfigurationFromMain = await executeCommand(`git show origin/${MAIN_BRANCH}:.gitlab-ci.yml`)
+  const currentStaging = /CURRENT_STAGING: (staging-.*)/g.exec(ciConfigurationFromMain)?.[1]
+  await executeCommand(`git fetch --no-tags origin ${currentStaging}`)
+
+  printLog(
+    `Checking if branch '${CI_COMMIT_REF_NAME}' (${CI_COMMIT_SHORT_SHA}) can be squash merged into ${currentStaging}...`
+  )
+
+  try {
+    await executeCommand(`git checkout ${MAIN_BRANCH} -f`)
+    await executeCommand(`git pull`)
+    await executeCommand(`git merge --squash "${CI_COMMIT_SHA}"`)
+    await executeCommand('git commit -am "squash test"')
```

Comment:
Puting this in a different try/catch with a different error message ("Failed to squash/merge PR onto main") could help understand what failed (is it the squash or the merge to staging)